Review and update as per the style guide

shavidissa opened the pull request:
Updated the following:
- Formatting of Kubernetes Concepts
- Ensure the page follows the style guide.

tengqm commented:
"service discovery" is a generic term which should not be changed to "Service discovery". In other words, the "service" in this term is not specific to the `Service` resource in kubernetes.

steveperry-53 commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -91,32 +91,32 @@ spec:
 
-This Service will also be assigned an IP address (sometimes called the "cluster IP"),
-which is used by the service proxies
+This Service also assigns an IP address (sometimes called the "cluster IP"),
```
The Service doesn't do the assigning.

Suggestion: This Service is assigned an IP address.
OR
Kubernetes assigns this Service and IP address.

sftim replied:
Behind the scenes, what's the controller doing the assigning?

AIUI sometimes it's something outside Kubernetes that assigns the Cluster IP. Have I got that right?

steveperry-53 commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -91,32 +91,32 @@ spec:
+protocol available via different port numbers.
 This offers a lot of flexibility for deploying and evolving your Services.
-For example, you can change the port number that pods expose in the next
+For example, you can change the port number the Pods are exposed in the next
```
This needs to be "that Pods expose". And we should probably say "numbers".

you can change the port numbers that Pods expose in the next version

sftim commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -5,7 +5,7 @@ title: Service
   title: Service discovery and load balancing
   description: >
-    No need to modify your application to use an unfamiliar service discovery mechanism. Kubernetes gives pods their own IP addresses and a single DNS name for a set of pods, and can load-balance across them.
+    No need to modify your application to use an unfamiliar Service discovery mechanism. Kubernetes gives Pods their own IP addresses and a single DNS name for a set of Pods, and can load-balance across them.
```
I think “service discovery” here is the wider sense, rather than talking about Service in the Kubernetes sense.

sftim commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -16,66 +16,66 @@ weight: 10
-No need to modify your application to use an unfamiliar service discovery mechanism.
-Kubernetes gives pods their own IP addresses and a single DNS name for a set of pods,
-:wqaand can load-balance across them.
+With Kubernetes you don't need to modify your application to use an unfamiliar Service discovery mechanism.
```
(and here)

steveperry-53 commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -235,19 +236,19 @@ By default, kube-proxy in userspace mode chooses a backend via a round-robin alg
-iptables rules which capture traffic to the Service's `clusterIP` (which is
-virtual) and `port` and redirects that traffic to one of the Service's
+iptables rules, which capture traffic to the Service's `clusterIP` and `port`,
+and redirects that traffic to one of the Service's
```
"and redirect that traffic"

sftim commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -16,66 +16,66 @@ weight: 10
 The Service abstraction enables this decoupling.
 
-### Cloud-native service discovery
+### Cloud-native Service discovery
```
Again, “service discovery” is the term here.

(When localized, I'd expect initial-caps Service to stay as the Kubernetes term, whereas “service discovery” might localize to, eg, обнаружения сервисов).

steveperry-53 commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -405,15 +407,15 @@ throughout your cluster then all Pods should automatically be able to resolve
-Namespace `"my-ns"`, the control plane and the DNS service acting together will
-create a DNS record for `"my-service.my-ns"`. Pods in the `"my-ns"` Namespace
+Namespace `"my-ns"`, the control plane and the DNS Service acting together
+creates a DNS record for `"my-service.my-ns"`. Pods in the `"my-ns"` Namespace
```
"create a DNS record"

shavidissa commented:
Hi @steveperry-53 and @sftim. Thank you so much for the comments :). I updated them. I made a terrible mistake when rebasing the commits. Now there is another commit, which is not mine and I tried several methods to remove it. But, it looks complicated and I don't want to make any weird changes and remove content from other PRs :( .

Therefore, I am creating a new PR that addresses all these comments. Can you take a look, please? I will assign you. 

[1] https://github.com/kubernetes/website/pull/15074

shavidissa commented:
> "service discovery" is a generic term which should not be changed to "Service discovery". In other words, the "service" in this term is not specific to the `Service` resource in kubernetes.

Thanks a lot for noticing the error @tengqm! I made a mistake with this PR and it got a bit messed up. Therefore, closed this an opened a new PR. Added you as an assignee too :)
